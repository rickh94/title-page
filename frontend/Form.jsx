import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus, faMinus } from '@fortawesome/free-solid-svg-icons'
import useInput from 'react-hanger/array/useInput'
import useArray from 'react-hanger/array/useArray'
import Swal from 'sweetalert2'

const fetchPost = async (endpoint, body) => {
  const response = await fetch(endpoint, {
    method: 'POST',
    body: JSON.stringify(body),
    headers: {
      'Content-Type': 'application/json',
    },
  })
  const data = await response.json()
  if (response.status >= 200 && response.status < 300) {
    return data
  } else {
    throw new Error(data.detail)
  }
}

export const Form = ({ setUrl, url }) => {
  const [[title], titleActions] = useInput('')
  const [[partName], partNameActions] = useInput('')
  const [[partAdditional], partAdditionalActions] = useInput('')
  const [composers, composersActions] = useArray([])
  const [extraLines, extraLinesActions] = useArray([])
  const [origFile, setFile] = useState(null)
  const [titlePageFileName, setTitlePageFileName] = useState('')

  const clear = () => {
    titleActions.clear()
    partNameActions.clear()
    partAdditionalActions.clear()
    composersActions.clear()
    extraLinesActions.clear()
  }

  const submit = async () => {
    const data = {
      title,
      composers,
      part: partName,
      extra_info: extraLines,
      part_additional: partAdditional,
    }
    try {
      const response = await fetchPost('/generate', data)
      setUrl(response.url)
      setTitlePageFileName(response.filename)
    } catch (e) {
      Swal.fire({
        title: 'Error',
        text: e.toString(),
        type: 'error',
      })
    }
  }

  const combine = async () => {
    const formData = new FormData()
    formData.append('title_page_filename', titlePageFileName)
    formData.append('file', origFile)
    try {
      const response = await fetch('/combine', {
        body: formData,
        method: 'POST',
      })
      const data = await response.json()
      if (response.status >= 200 && response.status < 300) {
        window.open(data.url, '_blank')
      } else {
        Swal.fire({
          title: 'Error',
          text: data.detail,
          type: 'error'
        })
      }
    } catch (e) {
      Swal.fire({
        title: 'Error',
        text: e.toString(),
        type: 'error',
      })
    }
  }

  return (
    <div className="column column-33">
      <label htmlFor="title">Title</label>
      <input
        type="text"
        name="title"
        id="title"
        placeholder="Symphony No. 5"
        value={title}
        onChange={titleActions.onChange}
        data-testid="title"
      />
      <label htmlFor="part">Part Name</label>
      <input
        type="text"
        name="part-name"
        id="part-name"
        placeholder="Violin I"
        value={partName}
        onChange={partNameActions.onChange}
        data-testid="part-name"
      />

      <label htmlFor="part-additional">Additional Part Information</label>
      <input
        type="text"
        name="part-additional"
        id="part-additional"
        placeholder="in Bb"
        value={partAdditional}
        onChange={partAdditionalActions.onChange}
        data-testid="part-additional"
      />
      <ListField
        items={composers}
        actions={composersActions}
        name="composer"
        label="Composers"
        placeholder="Ludwig van Beethoven"
      />
      <ListField
        items={extraLines}
        actions={extraLinesActions}
        name="extra-line"
        label="Extra Information Lines"
        placeholder="in C minor"
      />
      <button className="button" onClick={submit} data-testid="submit-button">
        Submit
      </button>
      <button className="button-outline" style={{ marginLeft: '1rem' }} onClick={clear}>
        Clear
      </button>
      <div style={{ paddingTop: '2rem' }} />

      {url && (
        <>
          <p>To add this title page to a file, upload it and click Combine</p>
          <label htmlFor="combine-file">Original File (Optional)</label>
          <input
            type="file"
            name="combine-file"
            id="combine-file"
            onChange={e => setFile(e.target.files[0])}
            data-testid="combine-file"
          />
          <button className="button" onClick={combine} data-testid="combine-button">
            Combine
          </button>
        </>
      )}
    </div>
  )
}

export const ListField = ({ items, actions, name, label, placeholder }) => {
  const [[next], nextActions] = useInput('')
  const appendItem = () => {
    actions.push(next)
    nextActions.clear()
  }

  const submitIfEnter = event => {
    const code = event.keyCode ? event.keyCode : event.which
    if (code === 13) {
      appendItem()
    }
  }

  return (
    <>
      <label htmlFor={`add-${name}`}>{label}</label>
      <div className="row" style={{ padding: '0 1rem' }}>
        <input
          type="text"
          name={`add-${name}`}
          id={`add-${name}`}
          value={next}
          onChange={nextActions.onChange}
          onKeyUp={submitIfEnter}
          placeholder={placeholder}
          data-testid={`${name}-next-input`}
        />
        <button
          className="button button-clear"
          onClick={appendItem}
          data-testid={`${name}-add-button`}
        >
          <FontAwesomeIcon icon={faPlus} />
        </button>
      </div>
      <ul id={`${name}s`} data-testid={`${name}-list`}>
        {items.map((item, idx) => (
          <li key={idx} data-testid={`${name}-item-${idx}`}>
            {item}
            <button
              className="button button-clear"
              onClick={() => actions.removeIndex(idx)}
              data-testid={`${name}-item-${idx}-remove`}
            >
              <FontAwesomeIcon icon={faMinus} />
            </button>
          </li>
        ))}
      </ul>
    </>
  )
}

ListField.propTypes = {
  items: PropTypes.array.isRequired,
  actions: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
}

Form.propTypes = {
  setUrl: PropTypes.func.isRequired,
  url: PropTypes.string,
}

export default Form
