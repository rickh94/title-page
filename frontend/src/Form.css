.combine-area {
  background-color: rgba(155, 77, 202, 0.2);
  padding: 1rem;
  border: 1px solid rgba(155, 77, 202, 0.4);
  text-align: center;
  display: flex;
  flex-direction: column;
  align-content: space-around;
  align-items: space-around;
  margin-bottom: 1rem;
}

.combine-area p {
  padding: 0;
  margin: 0;
  color: darkslategray;
}

.combine-area:hover {
  cursor: pointer;
}
