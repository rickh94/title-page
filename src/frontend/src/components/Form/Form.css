.combine-area {
  background-color: rgba(155, 77, 202, 0.2);
  padding: 1rem;
  border: 1px solid rgba(155, 77, 202, 0.4);
  text-align: center;
  display: flex;
  flex-direction: column;
  align-content: space-around;
  margin-bottom: 1rem;
}

.combine-area p {
  padding: 0;
  margin: 0;
  color: darkslategray;
}

.combine-area:hover {
  cursor: pointer;
}

.input-form {
  width: 100%
}

.button-row {
  padding: 0 1rem;
}


@media (min-width: 1280px) {
  .input-form {
    width: 32%;
    margin-right: 4rem;
  }
}
