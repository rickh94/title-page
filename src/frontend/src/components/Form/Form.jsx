import React, { useCallback, useState, Suspense } from 'react';
import PropTypes from 'prop-types';
import useInput from 'react-hanger/array/useInput';
import useArray from 'react-hanger/array/useArray';
import useBoolean from 'react-hanger/array/useBoolean';
import Swal from 'sweetalert2';
import { useDropzone } from 'react-dropzone';

import './Form.css';
// import FontSelect from '../FontSelect/FontSelect';
import ListField from '../ListField/ListField';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';

const FontSelect = React.lazy(() => import('../FontSelect/FontSelect'));

const fetchPost = async (endpoint, body) => {
  const response = await fetch(endpoint, {
    method: 'POST',
    body: JSON.stringify(body),
    headers: {
      'Content-Type': 'application/json',
    },
  });
  const data = await response.json();
  if (response.status >= 200 && response.status < 300) {
    return data;
  } else {
    throw new Error(data.detail);
  }
};

export const Form = ({ setUrl, url }) => {
  const [[title], titleActions] = useInput('');
  const [[partName], partNameActions] = useInput('');
  const [[partAdditional], partAdditionalActions] = useInput('');
  const [[font], fontActions] = useInput('Cormorant Garamond');
  const [composers, composersActions] = useArray([]);
  const [extraLines, extraLinesActions] = useArray([]);
  const [origFile, setFile] = useState(null);
  const [titlePageFileName, setTitlePageFileName] = useState('');
  const [composerDirty, composerDirtyActions] = useBoolean(false);
  const [extraLinesDirty, extraLinesDirtyActions] = useBoolean(false);
  const [showSelectFont, showSelectFontActions] = useBoolean(false);

  const clear = () => {
    titleActions.clear();
    partNameActions.clear();
    partAdditionalActions.clear();
    composersActions.clear();
    extraLinesActions.clear();
  };

  const submit = async () => {
    if (composerDirty || extraLinesDirty) {
      await Swal.fire({
        title: 'Data Entered',
        text: 'You have entered information in a list (composer or extra info) but not added it to the page. ' +
          'Please click the plus to add it or delete it from the input.',
        type: 'warning',
      });
      return;
    }
    const data = {
      title,
      composers,
      font,
      part: partName,
      extra_info: extraLines,
      part_additional: partAdditional,
    };
    try {
      const response = await fetchPost('/generate', data);
      setUrl(response.url);
      setTitlePageFileName(response.filename);
    } catch (e) {
      await Swal.fire({
        title: 'Error',
        text: e.toString(),
        type: 'error',
      });
    }
  };

  const onDrop = useCallback(acceptedFiles => {
    setFile(acceptedFiles[0]);
  }, []);
  const { getRootProps, getInputProps } = useDropzone({ onDrop });

  const combine = async () => {
    const formData = new FormData();
    formData.append('title_page_filename', titlePageFileName);
    formData.append('file', origFile);
    try {
      const response = await fetch('/combine', {
        body: formData,
        method: 'POST',
      });
      const data = await response.json();
      if (response.status >= 200 && response.status < 300) {
        window.open(data.url, '_blank');
      } else {
        Swal.fire({
          title: 'Error',
          text: data.detail,
          type: 'error',
        });
      }
    } catch (e) {
      Swal.fire({
        title: 'Error',
        text: e.toString(),
        type: 'error',
      });
    }
  };

  return (
    <div className="input-form" data-testid="form">
      <label htmlFor="title">Title</label>
      <input
        type="text"
        name="title"
        id="title"
        placeholder="Symphony No. 5"
        value={title}
        onChange={titleActions.onChange}
        data-testid="title"
      />
      <label htmlFor="part">Part Name</label>
      <input
        type="text"
        name="part-name"
        id="part-name"
        placeholder="Violin I"
        value={partName}
        onChange={partNameActions.onChange}
        data-testid="part-name"
      />

      <label htmlFor="part-additional">Additional Part Information</label>
      <input
        type="text"
        name="part-additional"
        id="part-additional"
        placeholder="in Bb"
        value={partAdditional}
        onChange={partAdditionalActions.onChange}
        data-testid="part-additional"
      />
      <ListField
        items={composers}
        actions={composersActions}
        name="composer"
        label="Composers"
        placeholder="Ludwig van Beethoven"
        dirtyActions={composerDirtyActions}
        completionsEndpoint="/completions/composers"
      />
      <ListField
        items={extraLines}
        actions={extraLinesActions}
        name="extra-line"
        label="Extra Information Lines"
        placeholder="in C minor"
        dirtyActions={extraLinesDirtyActions}
      />
      <div className="row button-row">
        <button
          className="button-clear px-0"
          title="Show select font"
          onClick={() => showSelectFontActions.toggle()}
          style={{ fontSize: '1.5rem' }}
        >
          Select Font{' '}
          <FontAwesomeIcon
            className="chevron-icon"
            icon={showSelectFont ? faChevronUp : faChevronDown}
          />
        </button>
      </div>
      {showSelectFont &&
      <Suspense fallback={<div>Loading...</div>}>
        <FontSelect value={font} valueActions={fontActions} />
      </Suspense>
      }
      <div className="row button-row mt-1">
        <button className="button" onClick={submit} data-testid="submit-button">
          Submit
        </button>
        <button
          className="button-outline"
          style={{ marginLeft: '1rem' }}
          onClick={clear}
          data-testid="clear-button"
        >
          Clear
        </button>

      </div>
      <div style={{ paddingTop: '2rem' }} />

      {url && (
        <>
          <p>To add this title page to a file, add it below and click Combine</p>
          {origFile && <p>Current File: {origFile.name}</p>}
          <div {...getRootProps()} className="combine-area" data-testid="combine-area">
            <input {...getInputProps()} />
            <p>Drag a file here or click to upload</p>
          </div>
          <button className="button" onClick={combine} data-testid="combine-button">
            Combine
          </button>
        </>
      )}
    </div>
  );
};

Form.propTypes = {
  setUrl: PropTypes.func.isRequired,
  url: PropTypes.string,
};

export default Form;
