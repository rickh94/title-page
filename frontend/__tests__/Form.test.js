import React from 'react'
import { render, fireEvent, cleanup, act } from '@testing-library/react'
import { Form, ListField } from '../Form'
import fetchMock from 'fetch-mock'
import Swal from 'sweetalert2'

afterEach(() => {
  cleanup()
  fetchMock.restore()
})

describe('Form', () => {
  it('matches snapshot', () => {
    const { container } = render(<Form url="" setUrl={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  it('submits form data', async done => {
    const setUrl = jest.fn()
    fetchMock.postOnce('/generate', {
      status: 200,
      body: {
        url: 'http://test.test',
        filename: 'test.pdf',
      },
    })
    const { getByTestId } = render(<Form url="" setUrl={setUrl} />)
    fireEvent.change(getByTestId('title'), { target: { value: 'Test Title' } })
    fireEvent.change(getByTestId('part-name'), { target: { value: 'Test Part Name' } })
    fireEvent.change(getByTestId('part-additional'), {
      target: { value: 'Additional Part Info' },
    })
    fireEvent.click(getByTestId('submit-button'))

    expect(fetchMock.lastOptions().body).toEqual(
      JSON.stringify({
        title: 'Test Title',
        composers: [],
        part: 'Test Part Name',
        extra_info: [],
        part_additional: 'Additional Part Info',
      })
    )
    await act(async () => await fetchMock.flush(true))
    expect(setUrl).toHaveBeenCalledWith('http://test.test')
    done()
  })

  it('alerts on submit error', async done => {
    const setUrl = jest.fn()
    const spy = jest.spyOn(Swal, 'fire')
    fetchMock.postOnce('/generate', {
      throws: new Error('Network Error'),
    })
    const { getByTestId } = render(<Form url="" setUrl={setUrl} />)
    fireEvent.change(getByTestId('title'), { target: { value: 'Test Title' } })
    fireEvent.change(getByTestId('part-name'), { target: { value: 'Test Part Name' } })
    fireEvent.change(getByTestId('part-additional'), {
      target: { value: 'Additional Part Info' },
    })
    fireEvent.click(getByTestId('submit-button'))

    expect(fetchMock.lastOptions().body).toEqual(
      JSON.stringify({
        title: 'Test Title',
        composers: [],
        part: 'Test Part Name',
        extra_info: [],
        part_additional: 'Additional Part Info',
      })
    )
    await act(async () => await fetchMock.flush(true))
    expect(spy).toHaveBeenCalledWith({
      title: 'Error',
      text: 'Error: Network Error',
      type: 'error',
    })
    done()
  })

  it('alerts on submit error response', async done => {
    const setUrl = jest.fn()
    const spy = jest.spyOn(Swal, 'fire')
    fetchMock.postOnce('/generate', {
      status: 400,
      body: {
        detail: 'Wrong Data',
      },
    })
    const { getByTestId } = render(<Form url="" setUrl={setUrl} />)
    fireEvent.change(getByTestId('title'), { target: { value: 'Test Title' } })
    fireEvent.change(getByTestId('part-name'), { target: { value: 'Test Part Name' } })
    fireEvent.change(getByTestId('part-additional'), {
      target: { value: 'Additional Part Info' },
    })
    fireEvent.click(getByTestId('submit-button'))

    expect(fetchMock.lastOptions().body).toEqual(
      JSON.stringify({
        title: 'Test Title',
        composers: [],
        part: 'Test Part Name',
        extra_info: [],
        part_additional: 'Additional Part Info',
      })
    )
    await act(async () => await fetchMock.flush(true))
    expect(spy).toHaveBeenCalledWith({
      title: 'Error',
      text: 'Error: Wrong Data',
      type: 'error',
    })
    done()
  })

  it('renders a file upload if a title page has already been generated', () => {
    const { queryByTestId } = render(<Form url="http://test.test" setUrl={jest.fn()} />)
    expect(queryByTestId('combine-file')).toBeTruthy()
    expect(queryByTestId('combine-button')).toBeTruthy()
  })

  it('calls combine correctly', async done => {
    fetchMock.postOnce('/generate', {
      status: 200,
      body: {
        url: 'http://test.test',
        filename: 'title.pdf',
      },
    })
    fetchMock.postOnce('/combine', {
      status: 200,
      body: {
        url: 'http://test.test2',
        filename: 'test.pdf',
      },
    })
    window.open = jest.fn()
    const setUrl = jest.fn()
    const { getByTestId } = render(<Form url="http://test.test" setUrl={setUrl} />)
    fireEvent.change(getByTestId('title'), { target: { value: 'Test Title' } })
    fireEvent.change(getByTestId('part-name'), { target: { value: 'Test Part Name' } })
    fireEvent.change(getByTestId('part-additional'), {
      target: { value: 'Additional Part Info' },
    })
    fireEvent.click(getByTestId('submit-button'))

    expect(fetchMock.lastOptions().body).toEqual(
      JSON.stringify({
        title: 'Test Title',
        composers: [],
        part: 'Test Part Name',
        extra_info: [],
        part_additional: 'Additional Part Info',
      })
    )
    await act(async () => await fetchMock.flush(true))

    const mockFile = new File(['one two three'], 'test-file.pdf', {
      type: 'application/pdf',
    })
    mockFile.filename = 'test-file.pdf'
    await fireEvent.change(getByTestId('combine-file'), {
      target: {
        files: [mockFile],
      },
    })
    await fireEvent.click(getByTestId('combine-button'))

    await act(async () => await fetchMock.flush(true))
    expect(fetchMock.lastOptions().body.get('file').filename).toEqual('test-file.pdf')
    expect(fetchMock.lastOptions().body.get('title_page_filename')).toEqual('title.pdf')
    expect(window.open).toHaveBeenCalledWith('http://test.test2', '_blank')
    done()
  })

  it('alerts if combine fails', async done => {
    fetchMock.postOnce('/generate', {
      status: 200,
      body: {
        url: 'http://test.test',
        filename: 'title.pdf',
      },
    })
    fetchMock.postOnce('/combine', {
      status: 500,
      body: {
        detail: 'Something has gone wrong'
      },
    })
    window.open = jest.fn()
    const spy = jest.spyOn(Swal, 'fire')
    const { getByTestId } = render(<Form url="http://test.test" setUrl={jest.fn()} />)
    fireEvent.change(getByTestId('title'), { target: { value: 'Test Title' } })
    fireEvent.change(getByTestId('part-name'), { target: { value: 'Test Part Name' } })
    fireEvent.change(getByTestId('part-additional'), {
      target: { value: 'Additional Part Info' },
    })
    fireEvent.click(getByTestId('submit-button'))
    await act(async () => await fetchMock.flush(true))

    const mockFile = new File(['one two three'], 'test-file.pdf', {
      type: 'application/pdf',
    })
    mockFile.filename = 'test-file.pdf'
    await fireEvent.change(getByTestId('combine-file'), {
      target: {
        files: [mockFile],
      },
    })
    await fireEvent.click(getByTestId('combine-button'))
    await act(async () => await fetchMock.flush(true))
    expect(spy).toHaveBeenCalledWith({
      title: 'Error',
      text: 'Something has gone wrong',
      type: 'error'
    })
    done()
  })

  it('alerts if combine thows an error', async done => {
    fetchMock.postOnce('/generate', {
      status: 200,
      body: {
        url: 'http://test.test',
        filename: 'title.pdf',
      },
    })
    fetchMock.postOnce('/combine', {
      throws: new Error('Network Error')
    })
    window.open = jest.fn()
    const spy = jest.spyOn(Swal, 'fire')
    const { getByTestId } = render(<Form url="http://test.test" setUrl={jest.fn()} />)
    fireEvent.change(getByTestId('title'), { target: { value: 'Test Title' } })
    fireEvent.change(getByTestId('part-name'), { target: { value: 'Test Part Name' } })
    fireEvent.change(getByTestId('part-additional'), {
      target: { value: 'Additional Part Info' },
    })
    fireEvent.click(getByTestId('submit-button'))
    await act(async () => await fetchMock.flush(true))

    const mockFile = new File(['one two three'], 'test-file.pdf', {
      type: 'application/pdf',
    })
    mockFile.filename = 'test-file.pdf'
    await fireEvent.change(getByTestId('combine-file'), {
      target: {
        files: [mockFile],
      },
    })
    await fireEvent.click(getByTestId('combine-button'))
    await act(async () => await fetchMock.flush(true))
    expect(spy).toHaveBeenCalledWith({
      title: 'Error',
      text: 'Error: Network Error',
      type: 'error'
    })
    done()
  })
})

describe('ListField', () => {
  it('render items and remove buttons for each item', () => {
    const mockActions = {
      push: jest.fn(),
      removeIndex: jest.fn(),
    }
    const { queryByTestId, getByTestId, container } = render(
      <ListField
        items={['one', 'two', 'three']}
        actions={mockActions}
        name="test-name"
        label="Label Test"
        placeholder="placeholder test"
      />
    )

    expect(container.querySelectorAll('li').length).toEqual(3)

    expect(queryByTestId('test-name-item-0')).toBeTruthy()
    expect(queryByTestId('test-name-item-1')).toBeTruthy()
    expect(queryByTestId('test-name-item-2')).toBeTruthy()

    expect(getByTestId('test-name-item-0').textContent).toEqual('one')
    expect(getByTestId('test-name-item-1').textContent).toEqual('two')
    expect(getByTestId('test-name-item-2').textContent).toEqual('three')

    expect(queryByTestId('test-name-item-0-remove')).toBeTruthy()
    expect(queryByTestId('test-name-item-1-remove')).toBeTruthy()
    expect(queryByTestId('test-name-item-2-remove')).toBeTruthy()
  })

  it('adds value from input on click', () => {
    const mockActions = {
      push: jest.fn(),
      removeIndex: jest.fn(),
    }
    const { getByTestId } = render(
      <ListField
        items={[]}
        actions={mockActions}
        name="test-name"
        label="Label Test"
        placeholder="placeholder test"
      />
    )
    fireEvent.change(getByTestId('test-name-next-input'), {
      target: { value: 'next value' },
    })
    fireEvent.click(getByTestId('test-name-add-button'))
    expect(mockActions.push).toHaveBeenCalledWith('next value')
  })

  it('removes item when remove button is clicked', () => {
    const mockActions = {
      push: jest.fn(),
      removeIndex: jest.fn(),
    }
    const { getByTestId, container } = render(
      <ListField
        items={['one', 'two', 'three']}
        actions={mockActions}
        name="test-name"
        label="Label Test"
        placeholder="placeholder test"
      />
    )

    expect(container.querySelectorAll('li').length).toEqual(3)

    fireEvent.click(getByTestId('test-name-item-0-remove'))
    expect(mockActions.removeIndex).toHaveBeenCalledWith(0)

    fireEvent.click(getByTestId('test-name-item-1-remove'))
    expect(mockActions.removeIndex).toHaveBeenCalledWith(1)

    fireEvent.click(getByTestId('test-name-item-2-remove'))
    expect(mockActions.removeIndex).toHaveBeenCalledWith(2)
  })
})
