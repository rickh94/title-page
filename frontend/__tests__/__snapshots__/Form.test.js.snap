// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot 1`] = `
<div>
  <div
    class="column column-33"
  >
    <label
      for="title"
    >
      Title
    </label>
    <input
      data-testid="title"
      id="title"
      name="title"
      placeholder="Symphony No. 5"
      type="text"
      value=""
    />
    <label
      for="part"
    >
      Part Name
    </label>
    <input
      data-testid="part-name"
      id="part-name"
      name="part-name"
      placeholder="Violin I"
      type="text"
      value=""
    />
    <label
      for="part-additional"
    >
      Additional Part Information
    </label>
    <input
      data-testid="part-additional"
      id="part-additional"
      name="part-additional"
      placeholder="in Bb"
      type="text"
      value=""
    />
    <label
      for="add-composer"
    >
      Composers
    </label>
    <div
      class="row"
      style="padding: 0px 1rem;"
    >
      <input
        data-testid="composer-next-input"
        id="add-composer"
        name="add-composer"
        placeholder="Ludwig van Beethoven"
        type="text"
        value=""
      />
      <button
        class="button button-clear"
        data-testid="composer-add-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <ul
      data-testid="composer-list"
      id="composers"
    />
    <label
      for="add-extra-line"
    >
      Extra Information Lines
    </label>
    <div
      class="row"
      style="padding: 0px 1rem;"
    >
      <input
        data-testid="extra-line-next-input"
        id="add-extra-line"
        name="add-extra-line"
        placeholder="in C minor"
        type="text"
        value=""
      />
      <button
        class="button button-clear"
        data-testid="extra-line-add-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <ul
      data-testid="extra-line-list"
      id="extra-lines"
    />
    <button
      class="button"
      data-testid="submit-button"
    >
      Submit
    </button>
    <button
      class="button-outline"
      style="margin-left: 1rem;"
    >
      Clear
    </button>
    <div
      style="padding-top: 2rem;"
    />
    
  </div>
</div>
`;
